src/elements/SingleReview.jsx

```diff
@@ -1,38 +1,49 @@
 import { useState, useEffect } from "react";
-import { patchReviewVotesById, getCommentsById } from "../utils/api";
+import {
+  patchReviewVotesById,
+  getCommentsById,
+  postCommentById,
+  getReviewById,
+} from "../utils/api";
 import { CommentCard } from "./CommentCard";
 import moment from "moment/moment";
+import { useParams } from "react-router-dom";
 
-export const SingleReview = ({
-  reviewData: {
-    review_id,
-    title,
-    owner,
-    created_at,
-    category,
-    designer,
-    review_img_url,
-    review_body,
-    votes,
-    comment_count,
-  },
-}) => {
-  const [localVotes, setLocalVotes] = useState(votes);
+export const SingleReview = ({ username }) => {
+  const { id } = useParams();
+
+  const [review, setReview] = useState();
+  const [reviewLoaded, setReviewLoaded] = useState(false);
+
+  const [localVotes, setLocalVotes] = useState(0);
   const [processingVote, setProcessingVote] = useState(false);
+
   const [comments, setComments] = useState([]);
   const [commentsLoaded, setCommentsLoaded] = useState(false);
+  const [postingComment, setPostingComment] = useState(false);
+  const [newCommentText, setNewCommentText] = useState("");
 
   useEffect(() => {
-    getCommentsById(review_id).then((responseComments) => {
-      setCommentsLoaded(true);
-      setComments(responseComments);
-    });
+    getReviewById(id)
+      .then((responseReview) => {
+        setLocalVotes(responseReview.votes);
+        setReview(responseReview);
+        setReviewLoaded(true);
+        return getCommentsById(responseReview.review_id);
+      })
+      .then((responseComments) => {
+        setCommentsLoaded(true);
+        setComments(responseComments);
+      })
+      .catch((err) => {
+        setReviewLoaded(true);
```

Comment:
if you have an error you'd ideally trigger a error state to change so you could give feedback to the user instead of changing the loading state (also setting loaded to true here wouldn't make sense because the data loading has failed). 

I guess the only danger is making sure you know if the comment or review API call failed so you set the right error state